src/GenAPI/Microsoft.DotNet.GenAPI/ISymbolExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiSymbolExtensions.Filtering;
+
+namespace Microsoft.DotNet.GenAPI
+{
+    internal static class ISymbolExtensions
+    {
+        /// <summary>
+        /// Determines if a symbol was generated by the compiler by checking for the presense of the CompilerGeneratedAttribute.
+        /// </summary>
+        /// <param name="symbol">Symbol to check</param>
+        /// <returns>True if the attribute was found.</returns>
+        public static bool IsCompilerGenerated(this ISymbol symbol)
+        {
+            return symbol.GetAttributes().Any(attribute =>
+                attribute?.AttributeClass?.Name == "CompilerGeneratedAttribute" &&
+                attribute.AttributeClass.ContainingNamespace.ToDisplayString().Equals("System.Runtime.CompilerServices", StringComparison.Ordinal));
+        }
```

Comment:
I think this might make sense to move into ApiSymbolExtensions' SymbolExtensions. If you want to keep it here, please remove the `I` prefix from the class name.